src/test/demultiplexer_node_test.cpp

```diff
@@ -0,0 +1,89 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../dl_node.hpp"
+#include "../logging.hpp"
+#include "../node.hpp"
+#include "test_utils.hpp"
+
+using namespace ovms;
+
+using testing::ElementsAre;
+
+class DemultiplexerDLNode : public DLNode {
```

Comment:
Should we have the same test for custom node after integration?

Reply:
I think full integration test will suffice - Custom node that will produce expected output shaped to later on demultiply. There is separate ticket for that.